flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDeserializationSchema.java

```diff
@@ -209,91 +213,157 @@ public int hashCode() {
 
 	// --------------------------------------------------------------------------------------------
 
-	private Object convert(JsonNode node, TypeInformation<?> info) {
-		if (info == Types.VOID || node.isNull()) {
-			return null;
-		} else if (info == Types.STRING) {
-			return node.asText();
-		} else if (info == Types.BOOLEAN) {
-			return Boolean.valueOf(node.asText().trim());
-		} else if (info == Types.BYTE) {
-			return Byte.valueOf(node.asText().trim());
-		} else if (info == Types.SHORT) {
-			return Short.valueOf(node.asText().trim());
-		} else if (info == Types.INT) {
-			return Integer.valueOf(node.asText().trim());
-		} else if (info == Types.LONG) {
-			return Long.valueOf(node.asText().trim());
-		} else if (info == Types.FLOAT) {
-			return Float.valueOf(node.asText().trim());
-		} else if (info == Types.DOUBLE) {
-			return Double.valueOf(node.asText().trim());
-		} else if (info == Types.BIG_DEC) {
-			return new BigDecimal(node.asText().trim());
-		} else if (info == Types.BIG_INT) {
-			return new BigInteger(node.asText().trim());
-		} else if (info == Types.SQL_DATE) {
-			return Date.valueOf(node.asText());
-		} else if (info == Types.SQL_TIME) {
-			return Time.valueOf(node.asText());
-		} else if (info == Types.SQL_TIMESTAMP) {
-			return Timestamp.valueOf(node.asText());
+	private interface RuntimeConverter extends Serializable {
+		Object convert(JsonNode node);
+	}
+
+	private static RuntimeConverter createRowRuntimeConverter(
+			RowTypeInfo rowTypeInfo,
+			boolean ignoreParseErrors,
+			boolean isTopLevel) {
+		final int rowArity = rowTypeInfo.getArity();
+		final TypeInformation<?>[] fieldTypes = rowTypeInfo.getFieldTypes();
+		final String[] fieldNames = rowTypeInfo.getFieldNames();
+
+		final RuntimeConverter[] fieldConverters = new RuntimeConverter[fieldTypes.length];
+		for (int i = 0; i < fieldTypes.length; i++) {
+			final TypeInformation<?> fieldType = fieldTypes[i];
+			final RuntimeConverter fieldConverter =
+				createNullableRuntimeConverter(fieldType, ignoreParseErrors);
+
+			fieldConverters[i] = (node) -> {
```

Comment:
Why not just `fieldConverters[i] = createNullableRuntimeConverter(fieldType, ignoreParseErrors);` ?

Reply:
We could simplify the logic a bit. I agree. I will put the checking into the method.